recipes/cuda-cccl-impl/meta.yaml

```diff
@@ -0,0 +1,64 @@
+{% set name = "cuda-cccl-impl" %}
+{% set version = "2.0.1" %}
+{% set thrust_version = "2.0.1" %}
+{% set cub_version = "2.0.1" %}
+{% set libcudacxx_version = "1.9.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.gz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/NVIDIA/thrust/archive/refs/tags/{{ thrust_version }}.{{ extension }}
+    folder: thrust
+    sha256: 06df373cf487cf3710be6798bc5441a03685b78e5c1338e2f14005f7ccd030e6  # [linux]
+    sha256: 39a9fc88b0b06da118cc3ed28cbfca27b2a7ae192877996cab1295176ceed65e  # [win]
+  - url: https://github.com/NVIDIA/cub/archive/refs/tags/{{ cub_version }}.{{ extension }}
+    folder: cub
+    sha256: beeef0962c87eca994db06cb9e9168a71c63fb369398a5b5f1db923140814701  # [linux]
+    sha256: 22308facf62421d6dd2489b9554ffa4908dca13d3921a3e7f2965c5a6984cd20  # [win]
+  - url: https://github.com/NVIDIA/libcudacxx/archive/refs/tags/{{ libcudacxx_version }}.{{ extension }}
+    folder: libcudacxx
+    sha256: 05b1435ad65f5bdef1bb8d1eb29dc8f0f7df34c01d77bf8686687a4603588bce  # [linux]
+    sha256: d7864460ce8def8e0620cb9dd891e94299ade75eac75aa96b34001af3cdb074c  # [win]
```

Comment:
We were thinking that we could have `cub` and `thrust` depend on this package

Reply:
Why? @kkraus14 mentioned that some projects rely on having a newer version of these packages than the cuda release.

Reply:
We eventually want to phase out those packages and were trying to decide on the best migration plan

Right they would be pulling in the latest version of CCCL

Reply:
Ah, forgot about that. Then you should do `1`.

Reply:
All good. Lot of moving parts here

Yeah we could add some `<` constraint on CCCL to the existing packages via a repodata patch to ensure they conflict

Does that seem reasonable or there other things we should consider?

Reply:
Why add repodata patch on existing packages, when you can add a run_constrained on this new package?

Reply:
Think this goes back to the point above about making new `cub` and `thrust` packages aliases to the CCCL. There are still packages using `cub` and `thrust` and we need a way to gradually move them over to CCCL

Reply:
@isuruf does this need to be addressed before merging?

Reply:
Yes, otherwise there'll be clobbering. You need to send a PR to do a repodata patch as @jakirkham said or add a `run_constrained` as I suggested. Your choice.

Reply:
I’ll add a `run_constrained` suggestion here.

Reply:
Suggestion for the constraint is here: https://github.com/conda-forge/staged-recipes/pull/21953/files#r1157595912